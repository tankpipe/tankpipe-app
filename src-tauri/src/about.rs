use serde::{Serialize, Deserialize};
use data_encoding::BASE64;


const VERSION: &str = env!("CARGO_PKG_VERSION");
const CONTACT: &str = "aHR0cHM6Ly9kaXNjb3JkLmdnL3V6cTg5R1ByVWc=";

#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq  )]
pub struct About {
    pub version: String,
    pub contact: String,
}

impl About {
    pub fn new() -> About {
        let input: Vec<u8> = CONTACT.into();
        let binding = BASE64.decode(&input).unwrap();
        let s = String::from_utf8_lossy(&binding);
        About{
            version: VERSION.to_string(),
            contact: s.into_owned()
        }
    }
}
