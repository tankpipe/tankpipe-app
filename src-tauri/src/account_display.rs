use accounts::account::{AccountType, Account};
use chrono::NaiveDate;
use rust_decimal::Decimal;
use serde::{Serialize, Deserialize};
use uuid::Uuid;
use accounts::serializer::*;
use rust_decimal_macros::dec;

#[derive(Clone, Serialize, Deserialize)]
pub struct NewAccount {
	pub name: String,
	pub account_type: AccountType,
	pub starting_balance: Decimal
}

impl NewAccount {
	pub fn to_account(self) -> Account {
		Account {
			id: Uuid::new_v4(),
			name: self.name,
			starting_balance: self.starting_balance,
			account_type: self.account_type,
            balance: dec!(0)
		}
	}

}


#[derive(Clone, Serialize, Deserialize)]
pub struct DateParam {
    #[serde(serialize_with = "serialize_naivedate")]
    #[serde(deserialize_with = "deserialize_naivedate")]
	pub date: NaiveDate
}

