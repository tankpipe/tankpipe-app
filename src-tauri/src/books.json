{   
    "accounts": {
      "44742ea3-f835-4267-b62e-0c1d4796d575": {
        "id": "44742ea3-f835-4267-b62e-0c1d4796d575",
        "name": "Savings Account 1",
        "account_type": "Debit",
        "balance": "0",
        "starting_balance": "0"
      },
      "66ac7995-4991-495a-b44e-d9882d579260": {
        "id": "66ac7995-4991-495a-b44e-d9882d579260",
        "name": "Credit Account 1",
        "account_type": "Credit",
        "balance": "0",
        "starting_balance": "0"
      }
    },
    "scheduled_transactions": [
      {
        "id": "8565ba58-2df8-4690-b4f1-7849b2b4b48f",
        "name": "Some income",
        "period": "Months",
        "frequency": 1,
        "start_date": "2022-06-04",
        "last_date": "2022-06-04",
        "amount": "200",
        "description": "Money in",
        "dr_account_id": "cb946148-79b0-43bc-a3b8-b8bd811aa1ad",
        "cr_account_id": "c77327ef-d27d-434c-a205-2e975e822c87"
      }
    ],
    "transactions": [
      {
        "id": "4ed6200e-4d96-4c40-b140-36de0cfc7bc8",
        "date": "2022-06-04",
        "description": "received moneys",
        "dr_account_id": "44742ea3-f835-4267-b62e-0c1d4796d575",
        "cr_account_id": "66ac7995-4991-495a-b44e-d9882d579260",
        "amount": "10000",
        "status": "Recorded"
      },
      {
        "id": "f70e1b5d-12ee-47b0-9fbb-ab7671ff4595",
        "date": "2022-06-05",
        "description": "Gave some moneys back",
        "dr_account_id": "66ac7995-4991-495a-b44e-d9882d579260",
        "cr_account_id": "44742ea3-f835-4267-b62e-0c1d4796d575",
        "amount": "98.99",
        "status": "Recorded"
      },
      {
        "id": "4ed6200e-4d96-4c40-b140-36de0cfc7bc9",
        "date": "2022-07-01",
        "description": "received more moneys",
        "dr_account_id": "44742ea3-f835-4267-b62e-0c1d4796d575",
        "cr_account_id": "66ac7995-4991-495a-b44e-d9882d579260",
        "amount": "250",
        "status": "Recorded"
      },
      {
        "id": "4ed6200e-4d96-4c40-b140-36de0cfc7bd1",
        "date": "2022-07-01",
        "description": "received more moneys",
        "dr_account_id": "44742ea3-f835-4267-b62e-0c1d4796d575",
        "cr_account_id": null,
        "amount": "150.45",
        "status": "Recorded"
      }
    ]
  }