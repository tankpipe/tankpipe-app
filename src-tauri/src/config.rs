use std::{ffi::{OsString, OsStr}, path::PathBuf};
use serde::{Deserializer, Serializer, Serialize, Deserialize};

#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq  )]
pub struct FileDetails {
    pub name: String,
    #[serde(serialize_with = "serialize_osstring")]
    #[serde(deserialize_with = "deserialize_osstring")]
    pub path: OsString,
}

impl FileDetails {
    pub fn empty() -> FileDetails {
        FileDetails{ name: "".to_string(), path: OsString::from("") }
    }

    pub fn from_path(name: &str, path: PathBuf) -> FileDetails {
        FileDetails{ name: name.to_string(), path: path.into_os_string() }
    }

    pub fn from_os_string(name: &str, path: OsString) -> FileDetails {
        FileDetails{ name: name.to_string(), path: path }
    }
}

#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq  )]
pub enum DateFormat {
    Locale,
    Regular,
    US,
    ISO,
}

#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq  )]
pub struct Config {
    pub version: String,
    #[serde(serialize_with = "serialize_osstring")]
    #[serde(deserialize_with = "deserialize_osstring")]
    pub data_dir: OsString,
    #[serde(serialize_with = "serialize_osstring")]
    #[serde(deserialize_with = "deserialize_osstring")]
    pub config_dir: OsString,
    pub last_file: FileDetails,
    pub recent_files: Vec<FileDetails>,
    pub display_date_format: DateFormat,
    pub import_date_format: String
}

impl Config {
    pub fn data_dir_as_os_str(&self) -> &OsStr {
        &self.data_dir.as_os_str()
    }

    pub fn settings_path(&self) -> PathBuf {
        PathBuf::from(self.config_dir.clone()).join("settings.json")
    }

    pub fn file_path(&self, name: &str) -> PathBuf {
        PathBuf::from(self.data_dir.clone()).join(name)
    }

    pub fn set_last_from_path(&mut self, path: OsString, name: &str) {
        let last_file: FileDetails;
        let os_string_path = OsString::from(path);
        self.last_file.path = os_string_path.clone();
        self.last_file.name = name.to_string();
        let index = self.recent_files.iter().position(|f| *f.path == os_string_path);
        match index {
            Some(i) => {
                last_file = self.recent_files.get(i).unwrap().clone();
                self.recent_files.remove(index.unwrap());
            },
            None =>last_file = FileDetails::from_os_string(name, os_string_path)
        }
        self.recent_files.insert(0, last_file);
    }

    pub fn set_last(&mut self, last_file: FileDetails) {
        if let Some(index) = self.recent_files.iter().position(|f| *f.path == last_file.path) {
            self.recent_files.remove(index);
        }
        self.last_file = last_file.clone();
        self.recent_files.insert(0, last_file);
    }

}

pub fn deserialize_osstring<'de, D>(deserializer: D) -> Result<OsString, D::Error>
    where D: Deserializer<'de>
{
    let path = String::deserialize(deserializer)?;
    Ok(OsString::from(path))
}

pub fn serialize_osstring<S>(path: &OsString, serializer: S) -> Result<S::Ok, S::Error>
    where S: Serializer,
{
    serializer.serialize_some(&PathBuf::from(path.as_os_str()))
}