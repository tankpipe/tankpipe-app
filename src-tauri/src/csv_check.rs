use serde::{Serialize, Deserialize};


use crate::reader::ColumnTypes;

#[derive(Clone, Serialize, Deserialize, Debug)]
pub struct CsvCheck {
    pub column_types: ColumnTypes,
    pub header_row: Vec<String>,
    pub sample_rows: Vec<Vec<String>>,
    pub mapping_exists: bool,
}

impl CsvCheck {
    pub fn create_new(column_types: ColumnTypes, header_row: Vec<String>, sample_rows: Vec<Vec<String>>) -> CsvCheck {
        CsvCheck {
            column_types: column_types,
            header_row: header_row,
            sample_rows: sample_rows,
            mapping_exists: false,
        }
    }

    pub fn set_mapping_exists(self, mapping_exists: bool) -> CsvCheck {
        CsvCheck {
            column_types: self.column_types,
            header_row: self.header_row,
            sample_rows: self.sample_rows,
            mapping_exists: mapping_exists,
        }
    }
}


