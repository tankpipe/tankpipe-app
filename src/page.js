import { writable } from 'svelte/store'

const views = {
    BOOKS: "BOOKS",
    ACCOUNTS: "ACCOUNTS",
    TRANSACTIONS: "TRANSACTIONS",
    SCHEDULES: "SCHEDULES",
    SETTINGS: "SETTINGS",
    NET_ASSETS: "NET_ASSETS",
    JOURNAL: "JOURNAL",
}

const modes = {
    NEW: "NEW",
    EDIT: "EDIT",
    MULTI_EDIT: "MULTI_EDIT",
    LIST: "LIST",
    LOAD: "LOAD"
}

const page = writable({view: views.ACCOUNTS, mode:modes.LIST, payload:{}})

const isEditMode = (page) => {
    return page.mode === modes.EDIT || page.mode === modes.NEW || page.mode === modes.MULTI_EDIT
}

const isSingleEditMode = (page) => {
    return page.mode === modes.EDIT || page.mode === modes.NEW
}

const isMultiEditMode = (page) => {
    return page.mode === modes.MULTI_EDIT
}

const isListMode = (page) => {
    return page.mode === modes.LIST
}

export {page, views, modes, isEditMode, isMultiEditMode, isSingleEditMode, isListMode}