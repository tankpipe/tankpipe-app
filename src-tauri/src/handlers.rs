
#![allow(dead_code)]

use accounts::books::Settings;
use accounts::{
    account::{Account, Entry},
    account::{Schedule, Transaction},
};
use std::ffi::OsString;
use std::{
    str::FromStr,
};
use uuid::Uuid;


use crate::BooksState;
use crate::account_display::{NewAccount, DateParam};
use crate::config::Config;
use crate::reader::load_transactions;



#[tauri::command]
pub fn transaction(
    state: tauri::State<BooksState>,
    transaction_id: Uuid,
) -> Result<Transaction, String> {
    println!("Fetching transaction {}", transaction_id);
    let mutex_guard = state.0.lock().unwrap();
    let x = mutex_guard.books.transaction(transaction_id);
    match x {
        Some(t) => Ok(t),
        None => Err(format!("Transaction {} not found", transaction_id)),
    }
}

#[tauri::command]
pub fn entries(state: tauri::State<BooksState>, account_id: Uuid) -> Result<Vec<Entry>, String> {
    println!("Fetching transactions for {}", account_id);
    let mutex_guard = state.0.lock().unwrap();
    let x = mutex_guard.books.account_entries(account_id);
    match x {
        Ok(_) => Ok(x.unwrap()),
        Err(e) => {
            println!("{}", e.error);
            Err(e.error)
        }
    }

}

#[tauri::command]
pub fn transactions(state: tauri::State<BooksState>, account_id: Uuid) -> Result<Vec<Transaction>, String> {
    println!("Fetching transactions for {}", account_id);
    let mutex_guard = state.0.lock().unwrap();
    let x = mutex_guard.books.account_transactions(account_id);
    match x {
        Ok(_) => Ok(x.unwrap()),
        Err(e) => {
            println!("{}", e.error);
            Err(e.error)
        }
    }

}

#[tauri::command]
pub fn add_transaction(
    state: tauri::State<BooksState>,
    mut transaction: Transaction,
) -> Result<(), String> {
    println!("Adding transaction {:?}", transaction);
    transaction.id = Uuid::new_v4();
    for mut e in transaction.entries.as_mut_slice() {
        e.id = Uuid::new_v4();
        e.transaction_id = transaction.id;
    }

    let mut mutex_guard = state.0.lock().unwrap();
    error_handler(mutex_guard.books.add_transaction(transaction))?;
    error_handler(mutex_guard.save())
}

#[tauri::command]
pub fn update_transaction(
    state: tauri::State<BooksState>,
    mut transaction: Transaction,
) -> Result<(), String> {
    println!("Updating transaction {:?}", transaction);
    for mut e in transaction.entries.as_mut_slice() {
        let zeros = Uuid::from_str(String::from("00000000-0000-0000-0000-000000000000").as_str()).unwrap();
        if e.id == zeros {
            e.id = Uuid::new_v4();
            e.transaction_id = transaction.id;
        }

    }
    let mut mutex_guard = state.0.lock().unwrap();
    error_handler(mutex_guard.books.update_transaction(transaction))?;
    error_handler(mutex_guard.save())
}

#[tauri::command]
pub fn delete_transaction(state: tauri::State<BooksState>, id: Uuid) -> Result<(), String> {
    println!("Deleting transaction {}", id);
    let mut mutex_guard = state.0.lock().unwrap();
    error_handler(mutex_guard.books.delete_transaction(&id))?;
    error_handler(mutex_guard.save())
}


#[tauri::command]
pub fn accounts(state: tauri::State<BooksState>) -> Vec<Account> {
    println!("Fetching accounts");
    let mutex_guard = state.0.lock().unwrap();
    mutex_guard.books.accounts()
}

#[tauri::command]
pub fn add_account(state: tauri::State<BooksState>, account: NewAccount) -> Result<(), String> {
    println!("Adding account {}", account.name);
    let mut mutex_guard = state.0.lock().unwrap();
    let _ = mutex_guard.books.add_account(account.to_account());
    error_handler(mutex_guard.save())
}

#[tauri::command]
pub fn update_account(state: tauri::State<BooksState>, account: Account) -> Result<(), String> {
    println!("Updating account {}", account.name);
    let mut mutex_guard = state.0.lock().unwrap();
    let _x = mutex_guard.books.add_account(account);
    error_handler(mutex_guard.save())
}

#[tauri::command]
pub fn delete_account(state: tauri::State<BooksState>, account: Account) -> Result<(), String> {
    println!("Deleting account {}", account.name);
    let mut mutex_guard = state.0.lock().unwrap();
    error_handler(mutex_guard.books.delete_account(&account.id))?;
    error_handler(mutex_guard.save())
}

#[tauri::command]
pub fn schedules(state: tauri::State<BooksState>) -> Vec<Schedule> {
    println!("Fetching scheduled transactions");
    let mutex_guard = state.0.lock().unwrap();
    mutex_guard.books.schedules().to_vec()
}

#[tauri::command]
pub fn add_schedule(state: tauri::State<BooksState>, mut schedule: Schedule) -> Result<(), String> {
    println!("Adding schedule: {}", schedule.name);
    schedule.id = Uuid::new_v4();
    for mut e in schedule.entries.as_mut_slice() {
        e.schedule_id = schedule.id;
    }
    let mut mutex_guard = state.0.lock().unwrap();
    error_handler(mutex_guard.books.add_schedule(schedule))?;
    error_handler(mutex_guard.save())
}

#[tauri::command]
pub fn update_schedule(state: tauri::State<BooksState>, schedule: Schedule) -> Result<(), String> {
    println!("Updating schedule {}", schedule.name);
    let mut mutex_guard = state.0.lock().unwrap();
    error_handler(mutex_guard.books.update_schedule(schedule))?;
    error_handler(mutex_guard.save())
}

#[tauri::command]
pub fn end_date(state: tauri::State<BooksState>) -> Option<DateParam> {
    let mutext_guard = state.0.lock().unwrap();

    match mutext_guard.books.end_date() {
        Some(d) => {
            println!("{}", d);
            Some(DateParam { date: d })
        }
        None => None,
    }
}

#[tauri::command]
pub fn generate(state: tauri::State<BooksState>, date: DateParam) -> Result<(), String> {
    println!("Generating to {}", date.date);
    let mut mutex_guard = state.0.lock().unwrap();
    mutex_guard.books.generate(date.date);
    error_handler(mutex_guard.save())
}

#[tauri::command]
pub fn update_settings(state: tauri::State<BooksState>, settings: Settings) -> Result<(), String> {
    println!("Updating settings: {:?}", settings);
    let mut mutex_guard = state.0.lock().unwrap();
    mutex_guard.books.settings = settings;
    error_handler(mutex_guard.save())
}

#[tauri::command]
pub fn settings(state: tauri::State<BooksState>) -> Settings {
    println!("Fetching settings");
    let mutex_guard = state.0.lock().unwrap();
    mutex_guard.books.settings.clone()
}

#[tauri::command]
pub fn config(state: tauri::State<BooksState>) -> Config {
    println!("Fetching config");
    let mutex_guard = state.0.lock().unwrap();
    mutex_guard.config.clone()
}

#[tauri::command]
pub fn load_csv(state: tauri::State<BooksState>, path: String, account: Account) -> Result<(), String> {
    println!("load_csv: {:?}, for account:{:?}", path, account.id);
    let load_result = load_transactions(path, &account);

    match load_result {
        Ok(transactions) => {
            let mut mutex_guard = state.0.lock().unwrap();
            for t in transactions {
                let add_result = mutex_guard.books.add_transaction(t);
                if add_result.is_err() {
                    return Err(add_result.unwrap_err().error);
                }
            }
            error_handler(mutex_guard.save())
        },
        Err(e) => Err(e.error),
    }
}

#[tauri::command]
pub fn load_file(state: tauri::State<BooksState>, path: String) -> Result<Vec<Account>, String> {
    println!("load_file: {:?}", path);
    let mut mutex_guard = state.0.lock().unwrap();
    let add_result = mutex_guard.load_books(&OsString::from(path));
    println!("{:?}", add_result);
    if add_result.is_err() {
        let the_error = add_result.unwrap_err().error;
        println!("{}", the_error);
        return Err(the_error);
    }
    Ok(mutex_guard.books.accounts())
}


#[tauri::command]
pub fn new_file(state: tauri::State<BooksState>, name: String) -> Result<Vec<Account>, String> {
    println!("new_file {}", name);
    let mut mutex_guard = state.0.lock().unwrap();
    let new_result = mutex_guard.new_books(&name);

    if new_result.is_err() {
        let the_error = new_result.err().unwrap().error;
        println!("{}", the_error);
        return Err(the_error);
    }
    Ok(mutex_guard.books.accounts())
}

fn error_handler(x: Result<(), accounts::books::BooksError>) -> Result<(), String> {
    match x {
        Ok(_) => Ok(()),
        Err(e) => {
            println!("{}", e.error);
            Err(e.error)
        }
    }
}

#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        let result = 2 + 2;
        assert_eq!(result, 4);
    }
}
