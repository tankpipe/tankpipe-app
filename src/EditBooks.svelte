<script>
    import {Errors} from './errors.js'
    import {onMount} from "svelte"
    import {page, modes, views} from "./page"
    import {accounts} from './accounts'
    import {context} from './context.js';
    import {emit} from '@tauri-apps/api/event'

    let msg = ""
    let errors = new Errors()
    let name
    let addButtonLabel = "Add"

    onMount(() => {
            addButtonLabel = "Add"
    })

    const onCancel = () => {
        page.set({view: views.ACCOUNTS, mode: modes.LIST})
    }

    const newFile = async (name) => {
        await invoke('new_file', {name: name}).then(loadFileSuccess, loadFileFailure)
    }

    function loadFileSuccess(result) {
        console.log(result)
        emit('file-loaded', "")
        accounts.set(result)
    }

    function loadFileFailure(result) {
        console.log(result)
        errors = new Errors()
        errors.addError("all", "We hit a snag: " + result)
    }

    const onAdd = () => {
        msg = ""
        errors = new Errors()
        if (!name || name.length < 1) {
             errors.addError("name", "Name is required")
        }

        const regex = /[^a-z0-9_\- ]/gi
        if (regex.test(name)) {
            errors.addError("name", "Name must be alphanumeric")
        }

        if (!errors.hasErrors()) {
            newFile(name)
        }

    }


</script>

{#if ! context.hasBooks}
    <div class="message">To get started give your first set of accounts a name. For example, Personal Finances.</div>
{/if}
<div class="form">
    <div class="form-heading">{$page.mode === modes.EDIT?"Edit":"New"} Books</div>
    <div class="form-row">
        <div class="widget">
            <label for="name">Name</label>
            <input id="name" class="description-input" class:error={errors.isInError("name")} bind:value={name}>
        </div>
    </div>
    <div class="form-button-row">
        <div class="msg-panel">
            {#each errors.getErrorMessages() as e}
            <p class="error-msg">{e}</p>
            {/each}
            {#if msg}
            <p class="success-msg">{msg}</p>
            {/if}
        </div>
        <div class="widget buttons">
            {#if context.hasBooks}
            <button on:click={onCancel}>Close</button>
            {/if}
            <button on:click={onAdd}>{addButtonLabel}</button>
        </div>
    </div>
</div>

<style>

    :global(.date-time-field input) {
        border: 1px solid #CCC !important;
        border-radius: 2px !important;
        height: 33px;
    }

    :root {
        --date-input-width: 110px;
    }

    .msg-panel {
        padding-left: 2px;
        font-size: 0.9em;
        float:left;
    }

    .msg-panel p {
        margin: 8px 0;
        max-width: 500px;
    }

    .error-msg {
        color: #FBC969;
    }

    .success-msg {
        color: green;
    }

    .error {
        border: 1px solid #FBC969 !important;
    }

    :global(.error-input input) {
        border: 1px solid #FBC969 !important;
    }

    .buttons {
        float: right;
        margin: 10px 12px 0 0;
    }

    .buttons button {
        min-width: 80px;
    }

    .form-row {
        display: inline-flex;
        float: left;
        width: 100%;
        clear:both;
    }

    .form-button-row {
        display: block;
        text-align: left;
    }

    .form-button-row {
        margin-left: 7px;
        margin-right: 2px;
    }

    input {
        margin-right: 0px;
    }

    .form {
        float: left;
        border-radius: 10px;
    }

    .widget {
        display: inline-block;
        padding: 5px 0px 5px 10px;
    }

    .description-input {
        width: 400px;
    }

</style>