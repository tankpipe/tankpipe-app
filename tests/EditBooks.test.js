import { render, waitFor } from '@testing-library/svelte'
import EditBooks from '../src/EditBooks.svelte'
import {page, views, modes} from '../src/page'
import { mockIPC } from "@tauri-apps/api/mocks"

it('is displayed correctly', async () => {
    page.set({view: views.EditBooks, mode: modes.EDIT})
    mockIPC((cmd, args) => {
        switch (cmd) {
            case "accounts": return account_data
        }

    })

    const {container} = render(EditBooks, {})
    expect(container.outerHTML).toMatchSnapshot();
});

