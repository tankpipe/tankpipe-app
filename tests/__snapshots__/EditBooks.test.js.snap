// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`is displayed correctly 1`] = `"<div><!----><div class="message">To get started give your first set of accounts a name. For example, Personal Finances.<p></p></div><!----> <div class="form svelte-og0nno"><div class="form-heading">Edit Books</div> <div class="form-row svelte-og0nno"><div class="widget svelte-og0nno"><label for="name">Name</label> <input id="name" class="description-input svelte-og0nno"></div></div> <div class="form-button-row svelte-og0nno"><div class="msg-panel svelte-og0nno"><!----> <!----></div> <div class="widget buttons svelte-og0nno"><!----> <button class="svelte-og0nno">Add</button></div></div></div></div>"`;
